controllers/routemonitor/routemonitor_supplement.go

```diff
@@ -254,3 +260,24 @@ func (r *RouteMonitorReconciler) getHostedControlPlane(namespace string) (hypers
 	}
 	return hcpList.Items[0], nil
 }
+
+// checkRedirect makes a HEAD request to the provided URL and handles any redirection responses.
+// If a redirection status code (300-399) is received, it updates the routeMonitor object's InsecureSkipTLSVerify field to true
+// and sets the extractedRouteURL to the redirection location.
```

Comment:
what does setting the extractedRouteURL mean? As far as I see this function only updates the `InsecureSkipTLSVerify` field on the routeMonitor?

Reply:
The purpose of checkRedirect() is to also update the extractedRouteURL value. The code has been updated accordingly.